src/main/kotlin/io/libp2p/core/multiformats/Multiaddr.kt

```diff
@@ -96,6 +105,36 @@ class Multiaddr(val components: List<Pair<Protocol, ByteArray>>) {
         )
     }
 
+    internal fun split(pred: (Protocol) -> Boolean): List<Multiaddr> {
+        val addresses = mutableListOf<Multiaddr>()
+        split(
+            addresses,
+            components,
+            pred
+        )
+        return addresses
+    }
+
+    private fun split(
+        accumulated: MutableList<Multiaddr>,
+        remainingComponents: List<Pair<Protocol, ByteArray>>,
+        pred: (Protocol) -> Boolean
+    ) {
+        val splitIndex = remainingComponents.indexOfLast { pred(it.first) }
```

Comment:
I suggest making an explicit type so that terms like `first` aren't ambiguous.

Reply:
Wouldn't do any harm, would it? I'll pick that up in a separate PR.